Monash-FIT3170/Finding-Neno
# Move reverse geocode to backend

Currently, when location information of missing reports and sightings is displayed, the location coordinates are passed into a reverse geocoding API call which is slow, and when refreshed it is done repeatedly (no caching). This PR removes this reverse geocoding process from the frontend and instead, the coordinates are reverse geocoded into the string information on the backend when a new missing report or sighting is to be added. This is done by using the python package geopy. 

Summary:
- geopy version 2.4.0 is added to the requirements.txt 
- new column called "location_string" is added to missing_reports and sightings tables in the database
- reverse geocode of coordinates to suburb information in frontend are removed
- python file called get_suburb.py which uses geopy is added which takes lat and long and returns the available corresponding information

To test this, 
1. Execute `pip install geopy` or alternatively `pip install -r requirements.txt` on root/src/backend
2. Run query  `ALTER table missing_reports ADD location_string varchar(255); ALTER table sightings ADD location_string varchar(255);` or run setup_db.py again (NOTE: this will reset db and delete existing data)
3. Create new missing report/sighting with location selected using the **"Search address"** option after typing in a location (this is because the drag feature is done on another PR that is pending for a review)
4. When report/sighting is created, check TablePlus and see if location_string is filled and also check the reports/sightings or dashboard page if the location information appears.
